torchtune/models/llama3/_tokenizer.py

```diff
@@ -166,11 +166,34 @@ def decode(
         Returns:
             str: The decoded string.
         """
-        return self.tt_model.decode(
-            token_ids,
-            truncate_at_eos=truncate_at_eos,
-            skip_special_tokens=skip_special_tokens,
-        )
+        if skip_special_tokens:
+            not_header = [True] * len(token_ids)
+            mask = True
+            idx = 0
+            while idx < len(token_ids):
```

Comment:
This feels like it would be more efficient with numpy arrays. What sucks is going list -> array -> list

But you could do something like:

```
indices_start = get_index_start_header(token_ids, start_header_id)
indices_end = get_index_start_header(token_ids, start_header_id)

for idx_start, index_end zip(indices_start, indices_end):
     mask[idx_start:index_end+1] = False
```

that way the python for-loop is over number of header ids, and not over indices